### ISSUE TITLE
socks5 proxy?

### ISSUE THREAD

**Issue by christeasdale** [TechPerplexed/Gooby#110]
I am looking to setup a sock5 proxy on my DigitalOcean droplet that is using Gooby.  I am no expert on all the magic that is working behind the scenes with Gooby so I thought I would ask.  I was looking at something like Dante or?  Is there a safe elegant way to do it with tools already installed with Gooby?  Better to set it up in a Docker container or on the Ubuntu host?  I use an addon in Firefox called SmartProxy which allows me to access specific websites with a proxy that my ISP Comcast is blocking.  I have been using random free proxy sites with not very much luck.  So I am not planning on pushing a bunch through the proxy.  I will of course password protect it.  I also did as a test a SSH tunnel through my Droplet to test things and it worked as expected to access blocked sites but not a very practical method day to day.  Any thoughts or ideas would be much appreciated.

**Comment by TechPerplexed** [comment]
Sorry for the somewhat late reply, we had to ask the real brain behind the Gooby project (Kelinger).

According to him it's not impossible. However, it would essentially mean a completely new reinvention of Gooby and its unified Docker experience. In the olden days of GooPlex, everything was non-Docker. That also meant that all the major media system releases were all non-Docker monstrosities and it wasn't unusual that countless server rebuilds were required whenever one or more components failed to work correctly.

What we can say (with a huge "this is unsupported" disclaimer) is that and I mostly quote my friend Kelinger here:

Docker specifically was written to run on one or more hosts.  You could, theoretically, purchase 10 $1.99 Docker instances from Scaleway for example and run each component on a dedicated instance. If DigitalOcean “droplets” work the same way, then why not.

Of course, our integration and deployment is based on a single Docker host running all containers so this would likely need to be adapted to that method of deployment. DO has an API and I would think it would go something like this (making up my own API commands for English reading and because I’m not going to develop this myself):
* Ocean droplet create instance plex.mydomain.com –type docker
* - Response:  created 123.45.67.89
* - Here’s the YAML for 123.45.67.89
* Ocean droplet create instance sonarr.mydomain.com –type docker
* - Response:  created 987.65.43.21
* - Here’s the YAML for 987.65.43.21
* Etc.

Now, you still have to deal with creating a common share (Rclone) that is seen by all those containers spread across multiple instances/servers/locations. No idea how to do this. The same goes for the proxy. Can it proxy something running on a different server? Probably. Is it the same config we’re using today? Probably not.

On the other hand:

If you are talking about setting up a Debian instance in Digital Ocean that is sized correctly, then your standard VPS requirements would apply. Specifically:

* Supports FUSE
* Supports Docker
* Has enough memory
* Has enough disk space
* Has enough bandwidth
* Doesn’t care about where or what you download
* Has CPU power to handle heavier tasks, if required, like transcoding, Usenet downloads, or multiple medium-to-high CPU intensive tasks (ie, don’t fall for the “8 core system!” advertising when it's 8 Atom cores than can’t unzip a single TV show in under 10 minutes).

So, in summary, yes it could probably be done, but no we won't support it nor assist in getting it to run on your system 😄

**Comment by christeasdale** [comment]
Thanks for the extended response, it is appreciated.  If anyone else has additional ideas or thoughts, please comment.
